src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -203,6 +256,11 @@ private static void WriteFunctionsMetricEvent(string executionId, ulong executio
                 MetricEventSource.Log.RaiseFunctionsMetricEvent(executionId, executionTimeSpan, executionCount, executionStage);
             }
 
+            private static string GetDictionaryKey(string name, Guid invocationId)
+            {
+                return string.Format("{0}_{1}", name, invocationId.ToString());
```

Comment:
Consider normalizing this ToLower, or setting your dictionary to be case insensitive